### PR TITLE
feat(blocks): Add code execution block

### LINKED ISSUE #8766: Create a Code Executor Block
Need to create a block that can execute code in a secure, isolated, sandboxed environment in the cloud.

We will use [this](https://e2b.dev/docs)

### PR DESCRIPTION
**Pull request by Abhi1992002**
- Resolves #8766 

Creates a block that executes code in an E2B sandbox.

Demo:

https://github.com/user-attachments/assets/460382c4-5bf7-4f96-a539-88ab263777de

### CODE REVIEW

File: autogpt_platform/frontend/src/components/integrations/credentials-input.tsx
Comment (Pwuts) [inline autogpt_platform/frontend/src/components/integrations/credentials-input.tsx:1]: Please put this in the right alphabetical location. Same for pyproject.toml, credentials-provider.tsx, types.ts.

File: autogpt_platform/backend/backend/blocks/code_executor.py [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+
+class CodeExecutionBlock(Block):
+    # TODO : Add support to upload and download files
+    # Currently, You can customized the CPU and Memory, only by creating a pre customized sandbox template
+    class Input(BlockSchema):
+        credentials: CredentialsMetaInput[
+            Literal["e2b_sandbox"], Literal["api_key"]
+        ] = CredentialsField(
+            provider="e2b_sandbox",
```
Comment (Pwuts) [inline autogpt_platform/backend/backend/blocks/code_executor.py]:
```suggestion
            provider="e2b",
```

File: autogpt_platform/backend/backend/blocks/code_executor.py [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+        )
+
+        language: ProgrammingLanguage = SchemaField(
+            description="Programming language to execute",
+            default=ProgrammingLanguage.PYTHON,
+            advanced=False,
+        )
+
+        timeout: int = SchemaField(
+            description="Sandbox timeout in seconds", default=300
```
Comment (Pwuts) [inline autogpt_platform/backend/backend/blocks/code_executor.py]:
```suggestion
            description="Execution timeout in seconds", default=300
```

File: autogpt_platform/backend/backend/blocks/code_executor.py [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+        )
+
+        template_id: str = SchemaField(
+            description="Create a pre Customized sandbox template. and Enter the template id here, Check more about it in here - https://e2b.dev/docs/sandbox-template",
```
Comment (Pwuts) [inline autogpt_platform/backend/backend/blocks/code_executor.py]:
```suggestion
            description=(
                "You can use an E2B sandbox template by entering its ID here. "
                "Check out the E2B docs for more details: "
                "[E2B - Sandbox template](https://e2b.dev/docs/sandbox-template)"
           	),
```

**Review by Pwuts — COMMENTED** [review]
This is AWESOME! I'll definitely be using this 😎

File: autogpt_platform/backend/backend/blocks/code_executor.py (hunk 1 above)
Comment (Pwuts) [reply autogpt_platform/backend/backend/blocks/code_executor.py]: Sorry if this was unclear: this also has to be updated everywhere else where `e2b_sandbox` is used.

File: autogpt_platform/backend/backend/blocks/code_executor.py [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+        credentials: CredentialsMetaInput[
+            Literal["e2b"], Literal["api_key"]
+        ] = CredentialsField(
+            provider="e2b",
+            supported_credential_types={"api_key"},
+            description="Enter your api key for the E2B Sandbox. You can get it in here - https://e2b.dev/docs",
+        )
+
+        # Todo : Option to run commond in background
+        commands: list[str] = SchemaField(
```
Comment (Pwuts) [inline autogpt_platform/backend/backend/blocks/code_executor.py]:
```suggestion
        setup_commands: list[str] = SchemaField(
```

File: autogpt_platform/backend/backend/blocks/code_executor.py [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+            description="Run these commands in the sandbox before running the code, You can use `curl` or `git` to install your desired debian based package manager. pip and npm is pre installed\n Currently, none of these commands work in the background.",
```
Comment (Pwuts) [inline autogpt_platform/backend/backend/blocks/code_executor.py]: Break up into <88 length lines + small reword:
```suggestion
            description=(
                "Shell commands to set up the sandbox before running the code. "
                "You can use `curl` or `git` to install your desired Debian based "
                "package manager. `pip` and `npm` are pre-installed.\n\n"
                "These commands are executed with `sh`, in the foreground."
            ),
```

File: autogpt_platform/backend/poetry.lock
Comment (Pwuts) [inline autogpt_platform/backend/poetry.lock:1]: Resolving the conflicts on this is impossible until #8885 is merged.

Comment (ntindle) [reply autogpt_platform/backend/poetry.lock:1]: merged!

**Review by Pwuts — DISMISSED** [review]

**Review by Pwuts — APPROVED** [review]

**Comment by Torantulino** [comment]
Hahaha that demo video is great 😆

**Pull request merged**